src/renderer/src/routes/pages/MigrationNeo3/MigrationNeo3ActionContent.tsx

```diff
@@ -0,0 +1,487 @@
+    } catch (error) {
+      console.error(error)
+
+      ToastHelper.error({ message: t('messages.getKeyError') })
+
+      handleGoBack()
+
+      return
+    }
+
+    let serviceAccount: Account<TBlockchainServiceKey>
+
+    try {
+      serviceAccount = AccountHelper.getServiceAccount({ account, key })
+
+      setData({ serviceAccount })
+    } catch (error) {
+      console.error(error)
+
+      ToastHelper.error({ message: t('messages.generateAccountError') })
+
+      handleGoBack()
+
+      return
+    }
+
+    let neo3Account: Account<'neo3'>
+
+    try {
+      neo3Account = bsAggregator.blockchainServicesByName.neo3.generateAccountFromKey(key) as Account<'neo3'>
+
+      setData({ neo3Account })
+    } catch (error) {
+      console.error(error)
+
+      ToastHelper.error({ message: t('messages.generateNeo3AccountError') })
+
+      handleGoBack()
+
+      return
+    }
+
+    await UtilsHelper.sleep(500)
+
+    setData({ isGeneratingAccounts: false })
+  }
+
+  const handleMigrateToNeo3 = async () => {
+    if (actionState.isActing || !serviceAccount || !neo3Account || !hasMigrationNeo3(service)) return
+
+    try {
+      const transactionHashes = await service.migrateToNeo3({ account: serviceAccount, address: neo3Account.address })
+
+      const doesNeo3AccountExist = doesAccountExist({
+        address: neo3Account.address,
+        blockchain: neo3Account.blockchain,
+      })
+
+      if (!doesNeo3AccountExist)
+        await importAccount({
+          address: neo3Account.address,
+          blockchain: neo3Account.blockchain,
+          key: neo3Account.key,
+          type: 'standard',
+          wallet: wallet!,
+        })
+
+      transactionHashes.forEach((hash, index) => {
+        const isGasToken = index === 0 && hasGasAmount
+
+        const transaction: TUseTransactionsTransfer = {
+          account,
+          amount: isGasToken ? gasTokenBalance!.amount! : neoTokenBalance!.amount,
+          asset: isGasToken ? gasToken!.symbol : neoToken!.symbol,
+          to: BSNeoLegacyConstants.MIGRATION_NEO3_COZ_ADDRESS,
+          from: account.address,
+          hash,
+          time: Date.now() / 1000,
+          fromAccount: account,
+          isPending: true,
+          toAccount: accounts.find(account => account.address === BSNeoLegacyConstants.MIGRATION_NEO3_COZ_ADDRESS),
+        }
+
+        dispatch(
+          authReducerActions.waitPendingTransaction({
+            transaction,
+            blockchainService: service,
+            network,
+            account: serviceAccount,
+          })
+        )
+      })
+
+      navigate(`/app/wallets/${account.id}/transactions`)
+
+      ToastHelper.success({ message: t('messages.migrationSuccess') })
+    } catch (error) {
+      console.error(error)
+
+      ToastHelper.error({ message: t('messages.migrationError') })
+    }
+  }
+
+  useEffect(() => {
+    if (!serviceAccount || !neo3Account) return
+
+    interval(handleCalculateToMigrateToNeo3Values, 8000)
```

Comment:
Why should you use an interval as the value never change ?

Reply:
Really... good catch.